src/App.tsx

```diff
@@ -8,35 +8,79 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
+import { User } from './types/User';
+import { getUsers } from './services/user';
+import { getUserPosts } from './services/post';
+import { Post } from './types/Post';
 
 export const App: React.FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
```

Comment:
does App.tsx really need all related to users state logic, I think we can easily move it to UserSelector and simplify the App.tsx